pkg/skaffold/initializer/build/init_test.go

```diff
@@ -0,0 +1,195 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"io"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/prompt"
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestDefaultInitializerGenerateManifests(t *testing.T) {
+	tests := []struct {
+		description       string
+		generatedInfos    []GeneratedArtifactInfo
+		mockPorts         []int
+		expectedManifests []string
+		force             bool
+		shouldErr         bool
+	}{
+		{
+			description: "one manifest, force",
+			generatedInfos: []GeneratedArtifactInfo{
+				{
+					ArtifactInfo{
+						ImageName: "image1",
+					},
+					"path/to/manifest",
+				},
+			},
+			expectedManifests: []string{
+				`apiVersion: v1
+kind: Service
+metadata:
+  name: image1
+  labels:
+    app: image1
+spec:
+  ports:
+  - port: 8080
+    protocol: TCP
+  clusterIP: None
+  selector:
+    app: image1
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: image1
+  labels:
+    app: image1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: image1
+  template:
+    metadata:
+      labels:
+        app: image1
+    spec:
+      containers:
+      - name: image1
+        image: image1
+`,
+			},
+			force: true,
+		},
+		{
+			description: "2 manifests, interactive",
+			generatedInfos: []GeneratedArtifactInfo{
+				{
+					ArtifactInfo{
+						ImageName: "image1",
+					},
+					"path/to/manifest1",
+				},
+				{
+					ArtifactInfo{
+						ImageName: "image2",
+					},
+					"path/to/manifest2",
+				},
+			},
+			mockPorts: []int{8080, 0},
+			expectedManifests: []string{
+				`apiVersion: v1
+kind: Service
+metadata:
+  name: image1
+  labels:
+    app: image1
+spec:
+  ports:
+  - port: 8080
+    protocol: TCP
+  clusterIP: None
+  selector:
+    app: image1
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: image1
+  labels:
+    app: image1
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: image1
+  template:
+    metadata:
+      labels:
+        app: image1
+    spec:
+      containers:
+      - name: image1
+        image: image1
+`,
+				`apiVersion: v1
+kind: Service
+metadata:
+  name: image2
+  labels:
+    app: image2
+spec:
+  clusterIP: None
+  selector:
+    app: image2
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: image2
+  labels:
+    app: image2
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: image2
+  template:
+    metadata:
+      labels:
+        app: image2
+    spec:
+      containers:
+      - name: image2
+        image: image2
+`,
+			},
+		},
+	}
+
+	for _, test := range tests {
+		testutil.Run(t, test.description, func(t *testutil.T) {
+			mockPortIdx := 0
+			t.Override(&prompt.PortForwardResourceFunc, func(_ io.Writer, imageName string) (int, error) {
+				port := test.mockPorts[mockPortIdx]
+				mockPortIdx++
```

Comment:
Can we add a test where mockPort returned is 0. IIUC, the generated manifest should not set anything.

Reply:
The second test showcases this functionality with one of the generated manifests. I can split it into its own test case if that's better

Reply:
ah. sorry i missed it before. you can update the in the test description to mention that. No need to add another test.